src/components/field.jsx

```diff
@@ -24,170 +20,55 @@ function fieldCounts(dimInfo, field) {
-    result += `Example values:\n${selected.map(rowToText).join(', ')}\n`;
-  }
-  if (excluded.length > 0) {
-    result += `\nExample of excluded values:\n${excluded.map(rowToText).join(', ')}`;
-  }
-
-  return result;
-}
-
-export function FieldWithoutState({
-  field, fieldData, layout, showFilterbox, model,
+export default function Field({
+  layout, field, fieldData, onCancelSelection,
 }) {
-  let classes = `field ${fieldData.qKeyType}`;
-
   if (!layout) {
-    return (
-      <div
-        className={classes}
-      >
-        <div className="name">
-          {field}
-        </div>
-      </div>
-    );
+    return null;
   }
 
   const total = layout.qListObject.qDimensionInfo.qCardinal;
   const states = layout.qListObject.qDimensionInfo.qStateCounts;
   const green = { width: `${Math.ceil((states.qSelected / total) * 100)}%` };
   const grey = { width: `${Math.ceil((states.qExcluded / total) * 100)}%` };
 
-  let descriptions = '';
-  if (states.qSelected) {
-    classes += ' filtered';
-  }
-
-  if (fieldData.qHasDuplicates) {
-    classes += ' has-duplicates';
-    descriptions += '. Duplicate values';
-  } else {
-    descriptions += '. Unique values';
-  }
-
-  if (fieldData.qHasNull) {
-    classes += ' has-null';
-    descriptions += ', has nulls';
-  } else {
-    descriptions += ', no nulls.';
-  }
-
-  const allExcluded = states.qExcluded === total;
-  const singleHit = states.qExcluded === total - 1;
-
-  if (allExcluded) {
-    classes += ' all-excluded';
-  }
-
-  if (singleHit) {
-    classes += ' single-hit';
-  }
-
-  const isSynthetic = (fieldData.qTags && fieldData.qTags.find(item => item === '$synthetic'));
-  if (isSynthetic) {
-    const syntheticFieldStyle = {
-      border: `2px dashed ${fieldData.backgroundColor}`,
-    };
-
-    return (
-      <div
-        className={classes}
-        style={syntheticFieldStyle}
-        title="Synthetic key"
-      >
-        <div className="name">
-          {field}
-        </div>
-      </div>
-    );
-  }
-
-  const fieldStyle = {
-    border: `2px solid ${fieldData.backgroundColor}`,
-  };
   return (
-    <div
-      className={classes}
-      style={fieldStyle}
-      title={`${states.qSelected} selected, ${states.qOption
-        + states.qAlternative} possible, ${
-        states.qExcluded
-      } excluded, total of ${total} values. ${descriptions}\n\n${firstFewValues(layout)}`}
-    >
-      <div className="name">
-        {field}
-      </div>
-      <div className="bartext">
-        {' '}
-        {fieldCounts(layout.qListObject.qDimensionInfo, fieldData)}
+    <div className="field">
+      <div className="innerContainer">
+        <div className="nameAndText">
+          <div className="name">
+            {field}
+          </div>
+          <div className="bartext">
+            {' '}
+            {fieldCounts(layout.qListObject.qDimensionInfo, fieldData)}
+          </div>
+        </div>
+        {onCancelSelection !== null
```

Comment:
I'd prefer if we keep logic outside of the templates, could we have this set in a variable and use the variable here instead?